src/components/ImageCard/CardIcon.jsx

```diff
@@ -16,6 +16,8 @@ const propTypes = {
 };
 
 /** This component calls out to our renderIconByName callback function with the icon name OR fails over to our legacy card icons from our legacy icon bundle */
+/* We test the implementation and do not want to trigger this console in our test logs */
+/* istanbul ignore next */
```

Comment:
Alternatively we can mock the console and expect the console to be called with a message that .includes(theWarningMessage)

One time we had the condition wrong for one of these warnings and we would've caught it if it was actually wrapped in a test.

Reply:
I see now we previously had mocked the console as I mention in the above comment and you refactored to remove and test the implementation. 

I'm fine not having a test around the `if (__DEV__ && !renderIconByName && !icons[icon]) {` condition but it's something we _can_ break and would potentially erroneously show the error. Might be worth a new issue for a broader discussion around if/when we test `warning()` usage (deprecations, errors like this, misconfigurations like in Breadcrumb with ResizeObserver polyfill, etc)